mantaray/backend.py

```diff
@@ -145,9 +145,20 @@ def _flush_and_close_socket(sock: _Socket) -> None:
 
 
 class IrcCore:
-    def __init__(self, server_config: config.ServerConfig, *, verbose: bool):
+    def __init__(self, settings: config.ServerSettings, *, verbose: bool):
+        self.settings = settings
         self._verbose = verbose
-        self._apply_config(server_config)
+
+        # The nick stored in self is the actual nick that the user has right now.
+        # The nick in settings represents what the user would like to have.
+        #
+        # These can be different if the server changes the user's nick to something
+        # like Guest1234. Also in the future we might handle nick name in use errors
+        # by trying f"{settings.nick}_" or similar.
+        self.nick = settings.nick
+
+        # Similar to nick this is where we are actually connected to.
```

Comment:
```suggestion
        # Similar to nick this is where we are actually connected to.
        # Most of the time this is same as settings.host, because we
        # reconnect shortly after changing the host in settings.
```